docs/spec_draft.md

```diff
@@ -1940,6 +1941,40 @@ produces a `result` tensor. For boolean tensors, it computes the logical NOT.
 
 [Back to Ops](#index-of-ops)
 
+
+## stablehlo.optimization_barrier
+
+### Semantics
+
+Blocks any optimization pass from moving computations across the barrier. It
+ensures that the operations producing the `operand` are executed before any
+operations that depend on `result`.
+
+### Arguments
+
+| Name      | Type                                                       |
+|-----------|------------------------------------------------------------|
+| `operand` | variadic number of tensors of any supported type or tokens |
+
+### Outputs
+
+| Name     | Type                                                       |
+|----------|------------------------------------------------------------|
+| `result` | variadic number of tensors of any supported type or tokens |
+
+### Constraints
+
+  * (C1) size(`operand`) $=$ size(`result`).
+  * (C2) type(`operand`) $=$ type(`result`).
+
```

Comment:
Why cant we write, operand = result

Reply:
`size` or `type` denote some of the static properties of these operands. `operand = result` might be confusing for someone to think that they have same values

Reply:
we can just say `for all i` and remove the $\in$ part.